src/main/java/io/anserini/search/SimpleImpactSearcher.java

```diff
@@ -429,48 +501,47 @@ public Map<String, Float> encode_with_onnx(String queryString) throws OrtExcepti
    * @throws OrtException if errors encountered during encoding
    * @return encoded query
    */
-  public String encode_with_onnx(Map<String, Float> queryWeight) throws OrtException {
+  public String encode_with_onnx(Map<String, Integer> queryWeight) throws OrtException {
     String encodedQ = "";
-    if (this.queryEncoder != null){
-      encodedQ = this.queryEncoder.generateEncodedQuery(queryWeight);
-    } else {
-      List<String> encodedQuery = new ArrayList<>();
-      for (Map.Entry<String, Float> entry : queryWeight.entrySet()) {
-        String token = entry.getKey();
-        Float tokenWeight = entry.getValue();
-        int weightQuanted = Math.round(tokenWeight);
-        for (int i = 0; i < weightQuanted; ++i) {
-          encodedQuery.add(token);
-        }
+    List<String> encodedQuery = new ArrayList<>();
+    for (Map.Entry<String, Integer> entry : queryWeight.entrySet()) {
+      String token = entry.getKey();
+      Integer tokenWeight = entry.getValue();
+      for (int i = 0; i < tokenWeight; ++i) {
+        encodedQuery.add(token);
       }
-      encodedQ = String.join(" ", encodedQuery);
     }
+    encodedQ = String.join(" ", encodedQuery);
+    
     return encodedQ;
   }
 
 
   /**
    * Searches the collection, returning 10 hits by default.
    *
-   * @param q query
+   * @param encoded_q query
    * @return array of search results
    * @throws IOException if error encountered during search
+   * @throws OrtException if error encountered during search
    */
-  public Result[] search(Map<String, Float> q) throws IOException, OrtException {
-    return search(q, 10);
+  public Result[] search(Map<String, Integer> encoded_q) throws IOException, OrtException {
+    return search(encoded_q, 10);
   }
 
   /**
    * Searches the collection.
    *
-   * @param q query
+   * @param encoded_q query
    * @param k number of hits
    * @return array of search results
    * @throws IOException if error encountered during search
+   * @throws OrtException if error encountered during search
    */
-  public Result[] search(Map<String, Float> q, int k) throws IOException, OrtException {
-    Query query = generator.buildQuery(Constants.CONTENTS, q);
-    String encodedQuery = encode_with_onnx(q);
+  public Result[] search(Map<String, Integer> encoded_q, int k) throws IOException, OrtException {
+    Map<String, Float> float_encoded_q = intToFloat(encoded_q);
```

Comment:
I'm still not very happy with this... we get a `Map<String, Integer>`... and then we covert to `Map<String, Float>`... then it comes back as a flat String.

Should we be redesigning the QueryEncoder?

Reply:
just saw this. this line is not for query encoder thing, its for QueryGenerator to build query, which takes in float weight. so here I just convert Int to Float, for example 10 -> 10.0